src/consensus/chainsel.md

```diff
@@ -0,0 +1,259 @@
+# Chain selection
+
+_Chain selection_ is the mechanism, specified in the Protocol associated to the
+current era in effect in the network, of identifying and and adopting the best
+chain in the network. A node might receive distinct chains from its peers. This
+situation is expected during normal operation of the Praos protocol and its
+derivatives.
+
+It is important to note that **all nodes** participating in the network are in
+charge of this responsibility, as opposed to producing blocks, which only some
+nodes will be involved in doing (namely those with forging credentials, the
+cryptographic keys for signing new blocks on behalf of a stake pool). This
+implies that even non-block-forging nodes contribute to the security of the
+network as a whole.
+
+Because each node performs chain selection, the Cardano network acts as a
+forward-filtering diffusion network.
+
+The block headers come from [`ChainSync`](./miniprotocols/chainsync.md) and
+the block bodies come from [`BlockFetch`](./miniprotocols/blockfetch.md).
+
+> [!NOTE]
```

Comment:
Do readers of the Blueprint need to be aware of the chain state? For the ledger state, I can imagine they do because the need to mention loading data from snapshots, fast rollbacks, and block applications.

Reply:
I guess they do in a way, since later on in the document, the chain state is referred to in the context of header validation.

Reply:
Yeah, I think it is something that we need to make explicit, because ChainSync uses headers, and nodes need to use chainsync